pkg/registry/core/pod/storage/eviction.go

```diff
@@ -220,7 +221,7 @@ func (r *EvictionREST) Create(ctx context.Context, name string, obj runtime.Obje
 		pdbName = pdb.Name
 
 		// If the pod is not ready, it doesn't count towards healthy and we should not decrement
-		if !podutil.IsPodReady(pod) && pdb.Status.CurrentHealthy >= pdb.Status.DesiredHealthy && pdb.Status.DesiredHealthy > 0 {
```

Comment:
remind me of the case where the pod is not healthy but we still want to add it to the disrupted pods list in the PDB and count it against DisruptionsAllowed? that seems to contradict the comment on this block

Reply:
Prior to the patch that introduced this comment (and associated logic), a disrupted pod (eg, pod is not ready) counted against your 'allowed disruptions.'  When trying to evict said pod, even if you had enough other healthy pods, you were unable to if your 'allowed disruptions' is 0.

Example:

I have 3 pods with PDB, MaxUnavailable==1.  All pods ready, pdb.status.AllowedDisruptions==1.  Pod A goes unready.  Now, pdb.status.AllowedDisruptions==0.  Try to evict Pod A, evict fails without patch.  With patch, evict succeeds.